### PR TITLE
Added CircleCI support

### PR DESCRIPTION
**Pull request by chrisfilo**
It will not catch failing examples until: https://github.com/sphinx-gallery/sphinx-gallery/issues/45 will be fixed.

### CODE REVIEW

**Comment by bthirion** [comment]
Do the CircleCI tests pass ?

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    - "~/nilearn_data"
+  override:
+    - pip install --upgrade pip
+    - pip install -e .
+    - pip install sphinx matplotlib coverage Pillow
+
+test:
+  override:
+    - make
+    - cd doc && make
```
Comment (lesteve) [inline circle.yml]: you make want to use make html here. It seems that make all does html-noplot. Full disclosure I always use make html ...

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  cache_directories:
+    - "~/nilearn_data"
+  override:
+    - pip install --upgrade pip
+    - pip install -e .
+    - pip install sphinx matplotlib coverage Pillow
+
+test:
+  override:
+    - make
```
Comment (lesteve) [inline circle.yml]: Also it seems like `make all` in nilearn is `make clean test doc-noplot` do you probably want to use `make clean test` instead.

**Comment by chrisfilo** [comment]
yup: https://circleci.com/gh/chrisfilo/nilearn/tree/enh%2Fcircleci

**Comment by chrisfilo** [comment]
Ready for merging.

**Comment by AlexandreAbraham** [comment]
:+1: Awesome!

**Comment by chrisfilo** [comment]
For every successful build the following products are available over the
web:
1. Full log:
https://circle-artifacts.com/gh/chrisfilo/nilearn/29/artifacts/0/home/ubuntu/log.txt
2. Coverage report:
https://circle-artifacts.com/gh/chrisfilo/nilearn/29/artifacts/0/home/ubuntu/nilearn/coverage/index.html
3. Current build of the documentation:
https://circle-artifacts.com/gh/chrisfilo/nilearn/29/artifacts/0/home/ubuntu/nilearn/doc/_build/html/index.html

On Thu, Jul 16, 2015 at 4:02 PM, Alexandre Abraham <notifications@github.com

**Comment by GaelVaroquaux** [comment]
:+1: for merge.

Two :+1:. I am merging.

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  override:
+    - make clean test test-coverage
+    # workaround - make html returns 0 even if examples fail to build
+    # (see https://github.com/sphinx-gallery/sphinx-gallery/issues/45)
+    - cat ~/log.txt && if grep -q "Traceback (most recent call last):" ~/log.txt; then false; else true; fi
+
+general:
+  artifacts:
+    - "doc/_build/html"
+    - "coverage"
```
Comment (lesteve) [inline circle.yml]: Not sure whether it needs to be in the artefact, but shouldn't it be .coverage (missing dot at the beginning) ?

Comment (GaelVaroquaux) [inline circle.yml]: > Not sure whether it needs to be in the artefact, but shouldn't it be .coverage
> (missing dot at the beginning) ?

It's the HTML report of coverage.

Comment (lesteve) [inline circle.yml]: Good point.

**Comment by lesteve** [comment]
Just wondering is there a way to remove stuff from the cache in CircleCI? Sometimes the datasets or the dataset downloading code and we need to rm -rf the dataset folder to download the dataset again.

**Comment by chrisfilo** [comment]
there is a button "rerun without cache"

**Comment by GaelVaroquaux** [comment]
Loic, Alex: create CircleCI account (go to the webpage), and we'll show
you the administration interface.

**Comment by lesteve** [comment]
Before I forget there are a few additional tweaks in our travis setup that we should try to move to CircleCI. One I am thinking of right now (I may add more later):

* running outside the git checkout in order to detect setup.py problems (e.g. missing data files)

**Comment by GaelVaroquaux** [comment]
>   • running outside the git checkout in order to detect setup.py problems (e.g.
>     missing data files)

I am not sure that this is important: travis does that for us. Actually,
I am quite happy that different strategies are taken on different
builders, as it exposes different bugs (eg #695)

**Comment by lesteve** [comment]
OK I wasn't sure whether we wanted to replace Travis by CircleCI in the long-term.

**Comment by GaelVaroquaux** [comment]
> OK I wasn't sure whether we wanted to replace Travis by CircleCI in the
> long-term.

No. Travis is going to be used for the fast stuff, and CircleCI for the
powerful stuff. In particular, we are severely relying on the
configuration matrix of travis.

**Pull request merged**